src/plugin.js

```diff
@@ -235,6 +233,11 @@ const onPlayerReady = (player, options) => {
  */
 const errors = function(options) {
   this.ready(() => {
+    if (options) {
```

Comment:
Building on what @dmlap said, you can simply leave the `isFlashSupported` where it is and modify the error message in `defaults` (i.e. changing `defaults.errors.4.message`), which seems to be the goal here. Then, if a user provides a custom message in `options`, it will override it automatically without all this boolean logic scattered around different functions.

In that case, the `isCustomError` flag goes away, this whole `if` block goes away, and all the changes in the `"error"` handler go away.